pkg/services/metrics/metrics.go

```diff
@@ -20,10 +20,14 @@ type Service struct {
 // Start runs http service with the exposed endpoint on the configured port.
 func (ms *Service) Start() {
 	if ms.config.Enabled {
-		ms.log.Info("service is running", zap.String("endpoint", ms.Addr))
-		err := ms.ListenAndServe()
-		if err != nil && !errors.Is(err, http.ErrServerClosed) {
-			ms.log.Warn("service couldn't start on configured port")
+		for _, srv := range ms.http {
+			go func() {
+				ms.log.Info("service is running", zap.String("endpoint", srv.Addr))
+				err := srv.ListenAndServe()
```

Comment:
And we can have `Listen` separated from `Serve` here the same way it is in the RPC server. Then an error could be returned from `Start` leading to startup failure (it's very unobvious now that failure to start some metric service is just logged, while RPC listener error is fatal).

Reply:
Fixed.